### ISSUE TITLE
object detection dataset_util.py potential bug

### ISSUE THREAD

**Issue by lwyjc** [tensorflow/models#4215]
Hi,

I have noticed that in the object detection repo, training data is not shuffled as expected. And in ```object_detection/utils/dataset_util.py``` I noticed that line 137 seems should be changed from the current ```records_dataset.shuffle(config.shuffle_buffer_size)``` to ```records_dataset = records_dataset.shuffle(config.shuffle_buffer_size)```. When I made the change locally, I see the shuffle to behave much better. Thanks.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by bitfort** [comment]
Adding some others.

**Comment by DefineFC** [comment]
I think this issue is fixed in #4205 😄

**Comment by harshini-gadige** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
